### PR TITLE
SP2023-167 row render h5p row

### PR DESCRIPTION
**Pull request by odacolquhoun**

### CODE REVIEW

File: h5p-row.js [hunk 1]
```diff
@@ -30,12 +30,20 @@ H5P.Row = (function (EventDispatcher) {
     var createHTML = function () {
       // Create wrapper
       wrapper = document.createElement('div');
+      wrapper.classList.add('h5p-row');
+
+      for (var i = 0; i < params.columns.length; i++) {
+        var column = document.createElement('div');
```
Comment (Camtrup) [inline h5p-row.js]: This should recieve some inline styling to fit the width and paddings that come with the forked column:

image

File: h5p-row.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ H5P.Row = (function (EventDispatcher) {
 
     // Wrapper element
     var wrapper;
-
+    console.log(params);
```
Comment (Camtrup) [inline h5p-row.js]: Should remove this

**Review by Camtrup — COMMENTED** [review]
Nicely done! One comment and it should be gucci💸. But done very elegantly!!!

File: h5p-row.js [hunk 3]
```diff
@@ -30,12 +30,31 @@ H5P.Row = (function (EventDispatcher) {
     var createHTML = function () {
       // Create wrapper
       wrapper = document.createElement('div');
+      wrapper.classList.add('h5p-row');
```
Comment (Languafe) [inline h5p-row.js]: Beware that the same class is added to the container inside the `attach` method. It's probably okay, but could potentially be confusing if trying to apply styles.

File: h5p-row.js [hunk 4]
```diff
@@ -30,12 +30,31 @@ H5P.Row = (function (EventDispatcher) {
...
 
-      // For now we just print the data :-)
-      const pre = document.createElement('pre');
-      pre.innerText = JSON.stringify(params, null, 2);
+      for (var i = 0; i < params.columns.length; i++) {
+        var column = document.createElement('div');
```
Comment (Languafe) [inline h5p-row.js]: I think `const` and `let` is cooler than `var`, but I don't know what the kids today think.

File: h5p-row.js [hunk 5]
```diff
@@ -30,12 +30,31 @@ H5P.Row = (function (EventDispatcher) {
...
+      wrapper.classList.add('h5p-row-wrapper');
 
-      // For now we just print the data :-)
-      const pre = document.createElement('pre');
-      pre.innerText = JSON.stringify(params, null, 2);
+      for (let i = 0; i < params.columns.length; i++) {
+        const column = document.createElement('div');
+        const columnData = params.columns[i];
+
+        column.classList.add('h5p-column');
```
Comment (Languafe) [inline h5p-row.js]: To match what H5P.Column does, add the class `h5p-row-content` here instead. H5P.Column itself will add the `h5p-column` class (but H5P.Row should not rely on this).

```suggestion
        column.classList.add('h5p-row-content');
```

File: h5p-row.css [hunk 6]
```diff
@@ -1,3 +1,12 @@
-.h5p-row {
-  background: salmon;
+.h5p-row-wrapper {
+  width: 100%;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-evenly;
 }
+
+.h5p-column {
```
Comment (Languafe) [inline h5p-row.css]: Change to match the suggestion below.
```suggestion
.h5p-row-content {
```

**Review by Languafe — CHANGES_REQUESTED** [review]
Super close, please review my suggestion for changing the class we add to the container for each column.

**Review by Languafe — APPROVED** [review]

**Pull request merged**